### PR TITLE
[CCXDEV-14053] Remove more log duplicates

### PR DESCRIPTION
**Pull request by juandspy**
# Description

Follow up on https://github.com/RedHatInsights/insights-results-aggregator/pull/2038 as we still have plenty of duplicates.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Testing steps

CI

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by JiriPapousek — APPROVED** [review]

File: server/dvo_handlers.go [hunk 1]
```diff
@@ -175,7 +175,14 @@ func (server *HTTPServer) getWorkloadsForNamespace(writer http.ResponseWriter, r
 
 	workload, err := server.StorageDvo.ReadWorkloadsForClusterAndNamespace(orgID, clusterName, namespaceID)
 	if err != nil {
-		log.Error().Err(err).Msg("Errors retrieving DVO workload recommendations from storage")
+		// err received at this point can be either TableNotFoundError (500) or ItemNotFoundError (404)
+		logLevel := log.Error
+		if _, ok := err.(*types.ItemNotFoundError); ok {
```
Comment (epapbak) [inline server/dvo_handlers.go:180]: I'd use `if errors.Is(err, types.ItemNotFoundError) {`

File: server/server.go [hunk 2]
```diff
@@ -547,8 +547,14 @@ func (server *HTTPServer) RuleClusterDetailEndpoint(writer http.ResponseWriter,
 	}
 
 	if err != nil {
-		log.Error().Err(err).Msgf("Unable to get list of clusters for specific rule %s", selector)
 		// err received at this point can be either TableNotFoundError (500) or ItemNotFoundError (404)
+		logLevel := log.Error
+		if _, ok := err.(*types.ItemNotFoundError); ok {
```
Comment (epapbak) [inline server/server.go:552]: same as above

File: server/dvo_handlers.go [hunk 3]
```diff
@@ -175,7 +175,14 @@ func (server *HTTPServer) getWorkloadsForNamespace(writer http.ResponseWriter, r
 
 	workload, err := server.StorageDvo.ReadWorkloadsForClusterAndNamespace(orgID, clusterName, namespaceID)
 	if err != nil {
-		log.Error().Err(err).Msg("Errors retrieving DVO workload recommendations from storage")
+		// err received at this point can be either TableNotFoundError (500) or ItemNotFoundError (404)
+		logLevel := log.Error
+		if errors.Is(err, &types.ItemNotFoundError{}) {
+			// If the item is not found, we shouldn't treat it as an error
+			logLevel = log.Warn
```
Comment (ikerreyes) [inline server/dvo_handlers.go:182]: Why not changing the log message as well?

**Review by epapbak — APPROVED** [review]

**Pull request merged**